src/utils/Terminal.php

```diff
@@ -108,42 +108,46 @@ protected static function getFallbackEscapeCodes() : void{
 		self::$COLOR_WHITE = "\x1b[38;5;231m";
 	}
 
+	protected static function executeShell(string $command) : string{
+		return (string) shell_exec($command);	
+	}
+
 	protected static function getEscapeCodes() : void{
-		self::$FORMAT_BOLD = `tput bold`;
-		self::$FORMAT_OBFUSCATED = `tput smacs`;
-		self::$FORMAT_ITALIC = `tput sitm`;
-		self::$FORMAT_UNDERLINE = `tput smul`;
+		self::$FORMAT_BOLD = self::executeShell('tput bold');
+		self::$FORMAT_OBFUSCATED = self::executeShell('tput smacs');
+		self::$FORMAT_ITALIC = self::executeShell('tput sitm');
+		self::$FORMAT_UNDERLINE = self::executeShell('tput smul');
 		self::$FORMAT_STRIKETHROUGH = "\x1b[9m"; //`tput `;
 
-		self::$FORMAT_RESET = `tput sgr0`;
+		self::$FORMAT_RESET = self::executeShell('tput sgr0');
 
 		$colors = (int) `tput colors`;
 		if($colors > 8){
-			self::$COLOR_BLACK = $colors >= 256 ? `tput setaf 16` : `tput setaf 0`;
-			self::$COLOR_DARK_BLUE = $colors >= 256 ? `tput setaf 19` : `tput setaf 4`;
-			self::$COLOR_DARK_GREEN = $colors >= 256 ? `tput setaf 34` : `tput setaf 2`;
-			self::$COLOR_DARK_AQUA = $colors >= 256 ? `tput setaf 37` : `tput setaf 6`;
-			self::$COLOR_DARK_RED = $colors >= 256 ? `tput setaf 124` : `tput setaf 1`;
-			self::$COLOR_PURPLE = $colors >= 256 ? `tput setaf 127` : `tput setaf 5`;
-			self::$COLOR_GOLD = $colors >= 256 ? `tput setaf 214` : `tput setaf 3`;
-			self::$COLOR_GRAY = $colors >= 256 ? `tput setaf 145` : `tput setaf 7`;
-			self::$COLOR_DARK_GRAY = $colors >= 256 ? `tput setaf 59` : `tput setaf 8`;
-			self::$COLOR_BLUE = $colors >= 256 ? `tput setaf 63` : `tput setaf 12`;
-			self::$COLOR_GREEN = $colors >= 256 ? `tput setaf 83` : `tput setaf 10`;
-			self::$COLOR_AQUA = $colors >= 256 ? `tput setaf 87` : `tput setaf 14`;
-			self::$COLOR_RED = $colors >= 256 ? `tput setaf 203` : `tput setaf 9`;
-			self::$COLOR_LIGHT_PURPLE = $colors >= 256 ? `tput setaf 207` : `tput setaf 13`;
-			self::$COLOR_YELLOW = $colors >= 256 ? `tput setaf 227` : `tput setaf 11`;
-			self::$COLOR_WHITE = $colors >= 256 ? `tput setaf 231` : `tput setaf 15`;
+			self::$COLOR_BLACK = (string) ($colors >= 256 ? `tput setaf 16` : `tput setaf 0`);
```

Comment:
why not using executeShell for this too?

Reply:
because this is complex expression, we don't need to force cast them to string twice